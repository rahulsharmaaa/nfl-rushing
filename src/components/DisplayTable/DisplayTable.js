import * as React from "react";
import { DataGrid, GridToolbar } from "@mui/x-data-grid";
import formatTableData from "../../utils/formatTableData";

export default function DisplayTable({ data }) {
  return (
    <div
      style={{ height: "90vh", width: "90vw", margin: "50px auto 50px auto" }}
      data-testid='mainDisplayTable'
    >
      <DataGrid
        {...formatTableData(data)}
        getRowId={(row) => row.Player + row.Team}
        components={{ Toolbar: GridToolbar }}
      />
    </div>
  );
}
