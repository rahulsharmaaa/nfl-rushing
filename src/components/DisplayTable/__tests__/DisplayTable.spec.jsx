import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { shallow, configure } from "enzyme";
import DisplayTable from "../DisplayTable";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
describe("Display Table", () => {
  it("renders the DisplayTable", () => {
    const props = {
      data: [
        {
          Player: "Joe Banyard",
          Team: "JAX",
          Pos: "RB",
          Att: 2,
          "Att/G": 2,
          Yds: 7,
          Avg: 3.5,
          "Yds/G": 7,
          TD: 0,
          Lng: "7",
          "1st": 0,
          "1st%": 0,
          "20+": 0,
          "40+": 0,
          FUM: 0
        }
      ]
    };
    const wrapper = shallow(<DisplayTable {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
