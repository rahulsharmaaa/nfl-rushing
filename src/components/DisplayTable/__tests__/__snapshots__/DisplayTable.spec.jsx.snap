// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Table renders the DisplayTable 1`] = `ShallowWrapper {}`;
